---
int16[] color_code
